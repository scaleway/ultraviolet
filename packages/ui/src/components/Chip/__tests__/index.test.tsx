import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithTheme, shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { Chip } from '..'

describe('checkbox', () => {
  test('renders correctly', () => shouldMatchEmotionSnapshot(<Chip>test</Chip>))
  test('renders correctly wiht icon', () =>
    shouldMatchEmotionSnapshot(
      <Chip>
        <Chip.Icon name="address" onClick={() => {}} />
        test
      </Chip>,
    ))

  test('renders correctly active', () =>
    shouldMatchEmotionSnapshot(
      <Chip active>
        test <Chip.Icon name="address" />
      </Chip>,
    ))

  test('renders correctly large', () =>
    shouldMatchEmotionSnapshot(
      <Chip size="large">
        test <Chip.Icon name="address" />
      </Chip>,
    ))

  test('renders correctly disabled', () =>
    shouldMatchEmotionSnapshot(
      <Chip disabled>
        test <Chip.Icon name="address" />
      </Chip>,
    ))

  test('renders correctly active disabled', () =>
    shouldMatchEmotionSnapshot(
      <Chip active disabled>
        test <Chip.Icon name="address" />
      </Chip>,
    ))

  test('throw error when using Chip.Icon outside of Chip', () => {
    expect(() => renderWithTheme(<Chip.Icon name="address" />)).toThrowError(
      'Chip.Icon can only be used inside a Chip component',
    )
  })

  test('renders correctly onClick', async () => {
    const mockOnClick1 = vi.fn()
    const mockOnClick2 = vi.fn()
    const mockOnClickIcon1 = vi.fn()
    const mockOnClickIcon2 = vi.fn()

    renderWithTheme(
      <>
        <Chip data-testid="test" onClick={mockOnClick1}>
          test
          <Chip.Icon
            data-testid="test-icon"
            name="arrowDown"
            onClick={mockOnClickIcon1}
          />
        </Chip>
        <Chip data-testid="test-disabled" disabled onClick={mockOnClick2}>
          test Disabled
          <Chip.Icon
            data-testid="test-icon-disabled"
            name="arrowDown"
            onClick={mockOnClickIcon2}
          />
        </Chip>
      </>,
    )

    const chip = screen.getByTestId('test')
    await userEvent.click(chip)
    expect(mockOnClick1).toHaveBeenCalledOnce()

    const chipDisabled = screen.getByTestId('test-disabled')
    await userEvent.click(chipDisabled)
    expect(mockOnClick2).toHaveBeenCalledTimes(0)

    const chipIcon = screen.getByTestId('test-icon')
    await userEvent.click(chipIcon)
    expect(mockOnClickIcon1).toHaveBeenCalledOnce()

    const chipIconDisabled = screen.getByTestId('test-icon-disabled')
    await userEvent.click(chipIconDisabled)
    expect(mockOnClickIcon2).toHaveBeenCalledTimes(0)
  })
})
