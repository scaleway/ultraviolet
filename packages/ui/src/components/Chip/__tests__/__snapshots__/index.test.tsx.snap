// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checkbox > renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="false"
        data-disabled="false"
        data-prominence="default"
        data-size="medium"
        data-trailing-icon="false"
      >
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkbox > renders correctly active 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

.emotion-2 {
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 0.25rem;
}

.emotion-2[data-has-onclick="true"][data-active="false"]:hover {
  background-color: #d9dadd;
}

.emotion-2[data-has-onclick="true"][data-active="true"]:hover {
  background-color: #8c40ef;
}

.emotion-2[data-disabled="true"] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="true"
        data-disabled="false"
        data-prominence="stronger"
        data-size="medium"
        data-trailing-icon="true"
      >
        test 
        <div
          class="emotion-2 emotion-3"
          data-active="true"
          data-disabled="false"
          data-has-onclick="false"
        >
          <svg
            class="styles__vbm0wq0 styles_prominence_stronger__vbm0wqi styles_disabled_false__vbm0wqf styles_sentiment_neutral__vbm0wq3 styles_size_small__vbm0wqa styles_undefined_compound_16__vbm0wq10"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M7.50285 14.8303L7.50684 14.8334L7.50875 14.8348C7.79522 15.0551 8.20448 15.0551 8.49148 14.8344L8.49286 14.8334L8.49685 14.8303L8.50974 14.8203C8.51744 14.8143 8.52733 14.8065 8.5393 14.7971C8.544 14.7933 8.54902 14.7894 8.55435 14.7851C8.59218 14.755 8.64587 14.7116 8.71314 14.6557C8.84761 14.5438 9.0368 14.3812 9.26247 14.1733C9.71281 13.7582 10.3137 13.1575 10.9164 12.4131C12.1039 10.9461 13.385 8.81416 13.385 6.3851C13.385 3.41099 10.974 1 7.99985 1C5.02574 1 2.61475 3.41099 2.61475 6.3851C2.61475 8.81416 3.89582 10.9461 5.08335 12.4131C5.68598 13.1575 6.28689 13.7582 6.73723 14.1733C6.9629 14.3812 7.15209 14.5438 7.28656 14.6557C7.35383 14.7116 7.40752 14.755 7.44535 14.7851C7.46427 14.8002 7.47923 14.8119 7.48996 14.8203L7.50285 14.8303ZM7.99985 8.00064C8.89208 8.00064 9.61538 7.27734 9.61538 6.3851C9.61538 5.49287 8.89208 4.76957 7.99985 4.76957C7.10762 4.76957 6.38432 5.49287 6.38432 6.3851C6.38432 7.27734 7.10762 8.00064 7.99985 8.00064Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkbox > renders correctly active disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

.emotion-2 {
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 0.25rem;
}

.emotion-2[data-has-onclick="true"][data-active="false"]:hover {
  background-color: #d9dadd;
}

.emotion-2[data-has-onclick="true"][data-active="true"]:hover {
  background-color: #8c40ef;
}

.emotion-2[data-disabled="true"] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="true"
        data-disabled="true"
        data-prominence="stronger"
        data-size="medium"
        data-trailing-icon="true"
      >
        test 
        <div
          class="emotion-2 emotion-3"
          data-active="true"
          data-disabled="true"
          data-has-onclick="false"
        >
          <svg
            class="styles__vbm0wq0 styles_prominence_stronger__vbm0wqi styles_disabled_true__vbm0wqe styles_sentiment_neutral__vbm0wq3 styles_size_small__vbm0wqa styles_undefined_compound_44__vbm0wq1s"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M7.50285 14.8303L7.50684 14.8334L7.50875 14.8348C7.79522 15.0551 8.20448 15.0551 8.49148 14.8344L8.49286 14.8334L8.49685 14.8303L8.50974 14.8203C8.51744 14.8143 8.52733 14.8065 8.5393 14.7971C8.544 14.7933 8.54902 14.7894 8.55435 14.7851C8.59218 14.755 8.64587 14.7116 8.71314 14.6557C8.84761 14.5438 9.0368 14.3812 9.26247 14.1733C9.71281 13.7582 10.3137 13.1575 10.9164 12.4131C12.1039 10.9461 13.385 8.81416 13.385 6.3851C13.385 3.41099 10.974 1 7.99985 1C5.02574 1 2.61475 3.41099 2.61475 6.3851C2.61475 8.81416 3.89582 10.9461 5.08335 12.4131C5.68598 13.1575 6.28689 13.7582 6.73723 14.1733C6.9629 14.3812 7.15209 14.5438 7.28656 14.6557C7.35383 14.7116 7.40752 14.755 7.44535 14.7851C7.46427 14.8002 7.47923 14.8119 7.48996 14.8203L7.50285 14.8303ZM7.99985 8.00064C8.89208 8.00064 9.61538 7.27734 9.61538 6.3851C9.61538 5.49287 8.89208 4.76957 7.99985 4.76957C7.10762 4.76957 6.38432 5.49287 6.38432 6.3851C6.38432 7.27734 7.10762 8.00064 7.99985 8.00064Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkbox > renders correctly disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

.emotion-2 {
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 0.25rem;
}

.emotion-2[data-has-onclick="true"][data-active="false"]:hover {
  background-color: #d9dadd;
}

.emotion-2[data-has-onclick="true"][data-active="true"]:hover {
  background-color: #8c40ef;
}

.emotion-2[data-disabled="true"] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="false"
        data-disabled="true"
        data-prominence="weak"
        data-size="medium"
        data-trailing-icon="true"
      >
        test 
        <div
          class="emotion-2 emotion-3"
          data-active="false"
          data-disabled="true"
          data-has-onclick="false"
        >
          <svg
            class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_true__vbm0wqe styles_sentiment_neutral__vbm0wq3 styles_size_small__vbm0wqa styles_undefined_compound_30__vbm0wq1e"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M7.50285 14.8303L7.50684 14.8334L7.50875 14.8348C7.79522 15.0551 8.20448 15.0551 8.49148 14.8344L8.49286 14.8334L8.49685 14.8303L8.50974 14.8203C8.51744 14.8143 8.52733 14.8065 8.5393 14.7971C8.544 14.7933 8.54902 14.7894 8.55435 14.7851C8.59218 14.755 8.64587 14.7116 8.71314 14.6557C8.84761 14.5438 9.0368 14.3812 9.26247 14.1733C9.71281 13.7582 10.3137 13.1575 10.9164 12.4131C12.1039 10.9461 13.385 8.81416 13.385 6.3851C13.385 3.41099 10.974 1 7.99985 1C5.02574 1 2.61475 3.41099 2.61475 6.3851C2.61475 8.81416 3.89582 10.9461 5.08335 12.4131C5.68598 13.1575 6.28689 13.7582 6.73723 14.1733C6.9629 14.3812 7.15209 14.5438 7.28656 14.6557C7.35383 14.7116 7.40752 14.755 7.44535 14.7851C7.46427 14.8002 7.47923 14.8119 7.48996 14.8203L7.50285 14.8303ZM7.99985 8.00064C8.89208 8.00064 9.61538 7.27734 9.61538 6.3851C9.61538 5.49287 8.89208 4.76957 7.99985 4.76957C7.10762 4.76957 6.38432 5.49287 6.38432 6.3851C6.38432 7.27734 7.10762 8.00064 7.99985 8.00064Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkbox > renders correctly large 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

.emotion-2 {
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 0.25rem;
}

.emotion-2[data-has-onclick="true"][data-active="false"]:hover {
  background-color: #d9dadd;
}

.emotion-2[data-has-onclick="true"][data-active="true"]:hover {
  background-color: #8c40ef;
}

.emotion-2[data-disabled="true"] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmall__m4c9own style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="false"
        data-disabled="false"
        data-prominence="default"
        data-size="large"
        data-trailing-icon="true"
      >
        test 
        <div
          class="emotion-2 emotion-3"
          data-active="false"
          data-disabled="false"
          data-has-onclick="false"
        >
          <svg
            class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_sentiment_neutral__vbm0wq3 styles_size_small__vbm0wqa styles_undefined_compound_2__vbm0wqm"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M7.50285 14.8303L7.50684 14.8334L7.50875 14.8348C7.79522 15.0551 8.20448 15.0551 8.49148 14.8344L8.49286 14.8334L8.49685 14.8303L8.50974 14.8203C8.51744 14.8143 8.52733 14.8065 8.5393 14.7971C8.544 14.7933 8.54902 14.7894 8.55435 14.7851C8.59218 14.755 8.64587 14.7116 8.71314 14.6557C8.84761 14.5438 9.0368 14.3812 9.26247 14.1733C9.71281 13.7582 10.3137 13.1575 10.9164 12.4131C12.1039 10.9461 13.385 8.81416 13.385 6.3851C13.385 3.41099 10.974 1 7.99985 1C5.02574 1 2.61475 3.41099 2.61475 6.3851C2.61475 8.81416 3.89582 10.9461 5.08335 12.4131C5.68598 13.1575 6.28689 13.7582 6.73723 14.1733C6.9629 14.3812 7.15209 14.5438 7.28656 14.6557C7.35383 14.7116 7.40752 14.755 7.44535 14.7851C7.46427 14.8002 7.47923 14.8119 7.48996 14.8203L7.50285 14.8303ZM7.99985 8.00064C8.89208 8.00064 9.61538 7.27734 9.61538 6.3851C9.61538 5.49287 8.89208 4.76957 7.99985 4.76957C7.10762 4.76957 6.38432 5.49287 6.38432 6.3851C6.38432 7.27734 7.10762 8.00064 7.99985 8.00064Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkbox > renders correctly wiht icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.25rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #ffffff;
  cursor: pointer;
  border: 1px solid #e9eaeb;
  text-align: center;
  color: #3f4250;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0[data-disabled="false"]:hover {
  background-color: #e9eaeb;
  border-color: #92959d;
  color: #222638;
}

.emotion-0[data-disabled="true"] {
  background-color: #f3f3f4;
  border-color: #d9dadd;
  color: #b5b7bd;
  cursor: not-allowed;
}

.emotion-0[data-active="true"] {
  background-color: #8c40ef;
  border-color: #8c40ef;
  color: #ffffff;
}

.emotion-0[data-active="true"][data-disabled="false"]:hover {
  background-color: #792dd4;
  border-color: #792dd4;
  color: #f9f9fa;
}

.emotion-0[data-active="true"][data-disabled="true"] {
  background-color: #e5dbfd;
  border: none;
}

.emotion-0[data-size='medium'] {
  height: 1.5rem;
  padding: 0.25rem 0.75rem;
}

.emotion-0[data-size='large'] {
  height: 2rem;
  padding: 0.25rem 1rem;
}

.emotion-0[data-trailing-icon="true"] {
  padding-right: 0.5rem;
}

.emotion-2 {
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  border-radius: 0.25rem;
}

.emotion-2[data-has-onclick="true"][data-active="false"]:hover {
  background-color: #d9dadd;
}

.emotion-2[data-has-onclick="true"][data-active="true"]:hover {
  background-color: #8c40ef;
}

.emotion-2[data-disabled="true"] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <div
      class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
    >
      <div
        class="emotion-0 emotion-1 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_center_xxsmall__toi52u5j"
        data-active="false"
        data-disabled="false"
        data-prominence="default"
        data-size="medium"
        data-trailing-icon="false"
      >
        <button
          class="emotion-2 emotion-3"
          data-active="false"
          data-disabled="false"
          data-has-onclick="true"
        >
          <svg
            class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_sentiment_neutral__vbm0wq3 styles_size_small__vbm0wqa styles_undefined_compound_2__vbm0wqm"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M7.50285 14.8303L7.50684 14.8334L7.50875 14.8348C7.79522 15.0551 8.20448 15.0551 8.49148 14.8344L8.49286 14.8334L8.49685 14.8303L8.50974 14.8203C8.51744 14.8143 8.52733 14.8065 8.5393 14.7971C8.544 14.7933 8.54902 14.7894 8.55435 14.7851C8.59218 14.755 8.64587 14.7116 8.71314 14.6557C8.84761 14.5438 9.0368 14.3812 9.26247 14.1733C9.71281 13.7582 10.3137 13.1575 10.9164 12.4131C12.1039 10.9461 13.385 8.81416 13.385 6.3851C13.385 3.41099 10.974 1 7.99985 1C5.02574 1 2.61475 3.41099 2.61475 6.3851C2.61475 8.81416 3.89582 10.9461 5.08335 12.4131C5.68598 13.1575 6.28689 13.7582 6.73723 14.1733C6.9629 14.3812 7.15209 14.5438 7.28656 14.6557C7.35383 14.7116 7.40752 14.755 7.44535 14.7851C7.46427 14.8002 7.47923 14.8119 7.48996 14.8203L7.50285 14.8303ZM7.99985 8.00064C8.89208 8.00064 9.61538 7.27734 9.61538 6.3851C9.61538 5.49287 8.89208 4.76957 7.99985 4.76957C7.10762 4.76957 6.38432 5.49287 6.38432 6.3851C6.38432 7.27734 7.10762 8.00064 7.99985 8.00064Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
